master_xml.py

```diff
@@ -308,6 +308,18 @@ def evaluateTokens(cards):
                         cards.get(tokenId)['released'] = True
                         card['related'].append(tokenId)
 
+                for template in ability.iter('TemplateId'):
+                    tokenId = template.attrib['V']
+                    token = cards.get(tokenId)
```

Comment:
Why can't this logic go in the loop at line 293?
Why is `TemplateId` capitalized?

Reply:
I see, so the files contain both `TemplateId` and `templateId`. How silly of them.

Reply:
Yepp. Just as an example:

Here `templateId` is used:
```
    <!--- - - - - - - - - - - - - -  PrizeWinningCow_Spawn - - - - - - - - - - - - - - -->
    <Ability id="PrizeWinningCow_Spawn" groupid="NONE" type="SpawnAndPlay">
      <Priority IntType="ConstantValue" V="400"/>
      <NumOfTimes IntType="ConstantValue" V="1"/>
      <SetTriggerInvalid V="False"/>
      <TriggeredIfDead V="False"/>
      <OnlyOneOfKindPerTrigger V="False"/>
      <UserInteraction V="None"/>
      <WillAffect V="Picked"/>
      <templateId V="200320"/>
      <NumCopies IntType="ConstantValue" V="1"/>
      <CopyData V="False"/>
      <TargetPlayer V="Same"/>
      <RowPlacement V="Location"/>
      <MoveLocation V="Inherit"/>
      <SpawnNextTo V="True"/>
      <SpawnTo V="Right"/>
      <BatchSpawns V="True"/>
      <AbilityTriggerConditions>
        <Trigger type="OnCardPowerChanged">
          <TriggerOwner V="Self"/>
          <ActiveLocation V="Active"/>
          <IgnoreSilence V="False"/>
          <PowerChangeType V="Damaged"/>
          <ChangeMode V="DamageAndBuff"/>
          <TypeOfChange V="Power"/>
          <FullCheckConditions CheckerList="AliveCards"/>
          <FullCheckConditionsCardRelated CheckerList="SourceCard"/>
        </Trigger>
        <Trigger type="OnCardWillMove">
          <TriggerOwner V="Self"/>
          <ActiveLocation V="Active"/>
          <IgnoreSilence V="False"/>
          <FromLocation V="16356"/>
          <FromBoardSide V="Ignore"/>
          <ToLocation V="16"/>
          <ToBoardSide V="Ignore"/>
          <FullCheckConditions CheckerList="AliveCards"/>
          <FullCheckConditionsCardRelated CheckerList="SourceCard"/>
        </Trigger>
      </AbilityTriggerConditions>
      <Conditions CheckerList="SourceCard"/>
    </Ability>
```

And here `TemplateId` is used:
```
    <!--- - - - - - - - - - - - - -  Draug_SpawnDraugir - - - - - - - - - - - - - - -->
    <Ability id="Draug_SpawnDraugir" groupid="NONE" type="SpawnEntities">
      <Priority IntType="ConstantValue" V="400"/>
      <NumOfTimes IntType="ConstantValue" V="1"/>
      <SetTriggerInvalid V="False"/>
      <TriggeredIfDead V="False"/>
      <OnlyOneOfKindPerTrigger V="False"/>
      <UserInteraction V="None"/>
      <WillAffect V="Picked"/>
      <AbilityTriggerConditions>
        <Trigger type="OnCardsAffectedByAbility">
          <TriggerOwner V="Self"/>
          <ActiveLocation V="Active"/>
          <IgnoreSilence V="False"/>
          <SourceAbility V="">
            <Item v="Draug_FilterEnemy"/>
          </SourceAbility>
          <FullCheckConditions CheckerList="AliveCards"/>
          <FullCheckConditionsCardRelated CheckerList="TriggerCards">
            <CardCountCondition>
              <CardCount IntType="CardCount" V="">
                <CardCountConditions CheckerList="AliveCards">
                  <BoardLocationCondition>
                    <CheckAgainst V="SourceCard"/>
                    <BoardLocation V="Graveyard"/>
                  </BoardLocationCondition>
                  <LoyaltyCondition IsNegated="True">
                    <CheckAgainst V="SourceCard"/>
                  </LoyaltyCondition>
                  <GroupCondition IsNegated="True">
                    <Group V="Gold"/>
                  </GroupCondition>
                </CardCountConditions>
              </CardCount>
              <CheckType V="GreaterOrEqual"/>
              <TargetValue IntType="ConstantValue" V="1"/>
            </CardCountCondition>
          </FullCheckConditionsCardRelated>
        </Trigger>
      </AbilityTriggerConditions>
      <Conditions CheckerList="AliveCards"/>
      <SpawnEntityList>
        <SpawnEntity>
          <TemplateId V="200457"/>
          <Side V="Opposite"/>
          <Location V="Graveyard"/>
          <Occupied V="Ignore"/>
          <NumberOfCopies IntType="ConstantValue" V="1"/>
        </SpawnEntity>
      </SpawnEntityList>
      <BanishEntityList/>
    </Ability>
```